File: VB/Core/Log4Net.Appender.LogViewVB.Core/Extensions/ServicesExtension.vb
Imports System.Runtime.CompilerServices
Imports LogViewerVB.Core
Imports Microsoft.Extensions.Configuration
Imports Microsoft.Extensions.DependencyInjection
Imports Microsoft.Extensions.Logging

Public Module ServicesExtension

    <Extension>
    Public Function AddLog4Net(builder As ILoggingBuilder, config As IConfiguration) As ILoggingBuilder

        builder _
            .ClearProviders() _
            .AddLog4Net(config.GetLog4NetConfiguration())

        Return builder

    End Function

    <Extension>
    Public Function AddLog4Net(builder As ILoggingBuilder, config As IConfiguration, configure As Action(Of DataStoreLoggerConfiguration)) As ILoggingBuilder

        builder.AddLog4Net(config)
        builder.Services.Configure(configure)
        Return builder

    End Function

    <Extension>
    Private Function GetLog4NetConfiguration(configuration As IConfiguration) As Log4NetProviderOptions

        Return configuration _
            .GetSection("Log4NetCore") _
            .Get(Of Log4NetProviderOptions)

    End Function

End Module

File: CSharp/Core/Log4Net.Appender.LogView.Core/Extensions/ServicesExtension.cs
using LogViewer.Core;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Logging;

namespace Log4Net.Appender.LogView.Core;

public static class ServicesExtension
{
    public static ILoggingBuilder AddLog4Net(this ILoggingBuilder builder, IConfiguration config)
        => builder
            .ClearProviders()
            .AddLog4Net(config.GetLog4NetConfiguration());

    public static ILoggingBuilder AddLog4Net(this ILoggingBuilder builder, IConfiguration config, Action<DataStoreLoggerConfiguration> configure)
    {
        builder
            .AddLog4Net(config)
            .Services.Configure(configure);

        return builder;
    }

    public static Log4NetProviderOptions? GetLog4NetConfiguration(this IConfiguration configuration)
        => configuration
            .GetSection("Log4NetCore")
            .Get<Log4NetProviderOptions>();
}